applications/adv_networking_bench/cpp/default_bench_op_tx.h

```diff
@@ -122,6 +117,9 @@ class AdvNetworkingBenchDefaultTxOp : public Operator {
       cudaMemset(gds_header_, 0, header_size_.get());
 
       populate_dummy_headers(pkt);
+      // ANO expects host order when setting
+      ip_src_ = ntohl(ip_src_);
+      ip_dst_ = ntohl(ip_dst_);
```

Comment:
can you explain why that's only done in gpu-only, and not cpu or HDS ?

Reply:
In GPU-only mode we need to construct the header one time on the host and put it in the correct order in memory before sending it out. In other modes the header is on the CPU and the ANO can do the swap for us. In reality a GPU kernel would do this if you were using GPU-only mode.